### DISCUSSION TITLE
Support for Musl

### DISCUSSION THREAD

**Discussion by kenaryn** [koka-lang/koka#328] [General]
Hello,

Is it planned to support for Musl? downloading the last release fails due to `obstack-printf` symbol not found. Like explained at https://www.gnu.org/software/gnulib/manual/html_node/obstack_005fvprintf.html, this is a function only surpported in Glibc.

With respect to building from source, 2 packages (*i.e.* `iscole-1.0.7` and `alex-3.2.6`) not supported in Musl cause the building to fail.
It would be marvelous if Koka maintainers would consider to support Musl as it is not a experimental library since many years. Void, Alpine and Chimera Linux are musl-based by default. Koka could benefit from a extended user base, while showing maturity and confidence in using it beyond a lab experiment or technological demonstration phase where direct competitors could not claim the same (*e.g.* Futhark, Unison, Eff, Effkt, Granule, Kind, ...).

**Comment by xialvjun** [comment]
But I have tried koka in an alpine docker container in windows host. 
Just `apk add libc-dev`.

**Comment by kenaryn** [comment]
Thank you for taking the time to answer :)

It indeed worked with `docker run --rm -it alpine /bin/sh; apk add libc-dev` but willing to learn and try out a language for months is not particuliarly an efficient use-case of a containerized applications, even though I could omit the `rm` flag and add a description DockerFile to make it more customized for a vaguely prepared coding environment.

I have rather adopted an another approach (thanks to `paper_`, a Void maintainer), *viz* a "chroot" environment with glibc running comparmentalized within the musl-based host environment. A bit trickier to set up than a dockerization repertoire but clearly a more permanent solution.

So I thank you again for bringing the docker idea to me which led me to the chroot-oriented solution. And now the Koka interactive compiler works as intended and that is a very pleasant news :)
